src/compiler/core.ts

```diff
@@ -1020,6 +1020,10 @@ namespace ts {
         }
 
         return "^(" + pattern + (usage === "exclude" ? ")($|/)" : ")$");
+    }
+
+    function replaceWildcardCharacters(component: string, singleAsteriskRegexFragment: string) {

```

Comment:
This still introduces a closure. Another approach would be:
```ts
// at :924
const singleAsteriskRegexFragmentForFiles = "([^./]*(\\.(?!min\\.js$))?)*";
const singleAsteriskRegexFragmentForOther = "[^/]*";

// at :938
const singleAsteriskRegexFragment = usage === "files" ? singleAsteriskRegexFragmentForFiles : singleAsteriskRegexFragmentForOther;
const replaceWildcardCharacter = usage === "files" ? replaceWildcardCharacterForFiles : replaceWildcardCharacterForOther;

// at :1000
subpattern += component.replace(reservedCharacterPattern, replaceWildcardCharacter); // uses local from :939

// at :1025
function replaceWildcardCharacterForFiles(match: string) {
    return replaceWildcardCharacter(match, singleAsteriskRegexFragmentForFiles);
}

function replaceWildcardCharacterForOther(match: string) {
    return replaceWildcardCharacter(match, singleAsteriskRegexFragmentForOther);
}

function replaceWildcardCharacter(match: string, singleAsteriskRegexFragment: string) {
    return match === "*" ? singleAsteriskRegexFragment : match === "?" ? "[^/]" : "\\" + match; 
}
```

With this, no new closures are introduced.